ecms_base/modules/custom/ecms_icon_library/src/Plugin/Field/FieldWidget/EcmsIconLibraryWidget.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_icon_library\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+
+/**
+ * Plugin implementation of the 'field_example_text' widget.
+ *
+ * @FieldWidget(
+ *   id = "ecms_icon_library_widget",
+ *   module = "ecms_icon_library",
+ *   label = @Translation("Icon library"),
+ *   field_types = {
+ *     "ecms_icon_library"
+ *   }
+ * )
+ */
+class EcmsIconLibraryWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state): array {
+
+    $element['pl_icon'] =
+    [
+      '#type' => 'select',
+      '#title' => 'Icon',
+      '#description' => t('Select an icon from the dropdown. A preview of the icon will be shown below after you select it.'),
+      '#options' => $this->getIconOptions(),
+      // Do not display a 'multiple' select box if there is only one option.
+      '#default_value' => $items[$delta]->get('pl_icon')->getValue(),
+      '#field_suffix' => '<div id="qh-icon-field" class="qh__icon-library-preview"></div>',
```

Comment:
If there is more than one icon added to the page, I think there would be duplicate "id" attributes which might wreak havoc with the AJAX callback. Or is this the icon assignment for the media item, so there never *would* be more than one on the page?